backend/src/main/java/c/team/session/statistics/SessionAnswersService.java

```diff
@@ -0,0 +1,36 @@
+package c.team.session.statistics;
+
+import c.team.quiz.QuestionRepository;
+import c.team.quiz.exception.QuestionNotFoundException;
+import c.team.quiz.model.Question;
+import c.team.session.statistics.model.SessionAnswers;
+import c.team.session.statistics.model.SessionAnswersDto;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@AllArgsConstructor
+public class SessionAnswersService {
+
+    private final QuestionRepository questionRepository;
+    private final SessionAnswersRepository sessionAnswersRepository;
+
+    public synchronized void addAnswers(String sessionId, String questionId, List<Integer> answers) {
+        SessionAnswers sessionAnswers = sessionAnswersRepository.findBySessionIdAndQuestionId(sessionId, questionId).orElseGet(() -> new SessionAnswers(sessionId, questionId, new HashMap<>()));
```

Comment:
To potencjalnie można by bardziej wertykalnie przepisać (by nie było w jednej linijce - entery przed kropkami czy coś), ale nie jest to specjalnie problemem.

Reply:
ok, zmieniłem